numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3687,6 +3694,177 @@ static void
+6. After that we take care of the unaligned portion by simply looping over it, like the previous algorithm was doing.
+7. For a simple benchmark, this algorithm gave around 40-50% performance boost.
+*/
+
+
+ 
+#if HAVE_ATTRIBUTE_TARGET_AVX2
+static void
+INT_fastclip(npy_int *in, npy_intp ni, npy_int *min, npy_int *max, npy_int *out)
+{
+    npy_intp i;
+    npy_int max_val = 0, min_val = 0;
+    int unaligned = ni%8;
+    __m128i vec_max,vec_min;
+    __m256i vec_max_256,vec_min_256,vec_array;
+
+    if (max != NULL) {
+       max_val = *max;
+       vec_max = _mm_set1_epi32(max_val);
+       vec_max_256 = _mm256_broadcastd_epi32(vec_max);
+    }
+    if (min != NULL) {
+        min_val = *min;
+        vec_min = _mm_set1_epi32(min_val);
+        vec_min_256 = _mm256_broadcastd_epi32(vec_min);
+	
+    }
+    if (max == NULL) {
+	for(i=0;i<ni/8;i++){
+           vec_array = _mm256_loadu_si256(in+(i*8));
+           vec_array = _mm256_max_epi32(vec_array,vec_min_256);
+           _mm256_storeu_si256(out+(i*8),vec_array);
+       }
+       for(npy_int j=0;j<unaligned;j++){
+             if (in[i*8+j]<min_val) {
+                out[i*8+j]   = min_val;
+            }
+	}
+    	
+    }
+    else if (min == NULL) {
+	for(i=0;i<ni/8;i++){
+       vec_array = _mm256_loadu_si256(in+(i*8));
+       vec_array = _mm256_min_epi32(vec_array,vec_max_256);
+       _mm256_storeu_si256(out+(i*8),vec_array);
+   }
+       for(npy_int j=0;j<unaligned;j++){
+             if (in[i*8+j]>max_val) {
+                out[i*8+j]   = max_val;
+            }
+	}
+   }
+   else {
+  	for(i=0;i<ni/8;i++){
+           vec_array = _mm256_loadu_si256(in+(i*8));
+           vec_array = _mm256_max_epi32(vec_array,vec_min_256);
+           vec_array = _mm256_min_epi32(vec_array,vec_max_256);
+           _mm256_storeu_si256(out+(i*8),vec_array);
+       }
+       for(npy_int j=0;j<unaligned;j++){
+            if (in[i*8+j]<min_val) {
+                    out[i*8+j]   = min_val;
+                }
+                else if (in[i*8+j]>max_val) {
+                    out[i*8+j]   = max_val;
+                }
+
+       }
+    }
+}
+/**begin repeat
+ *
+ * #name = BOOL,
+ *         BYTE, UBYTE, SHORT, USHORT,UINT,
+ *         LONG, ULONG, LONGLONG, ULONGLONG,
+ *         HALF, FLOAT, DOUBLE, LONGDOUBLE,
+ *         DATETIME, TIMEDELTA#
+ * #type = npy_bool,
+ *         npy_byte, npy_ubyte, npy_short, npy_ushort,npy_uint,
+ *         npy_long, npy_ulong, npy_longlong, npy_ulonglong,
+ *         npy_half, npy_float, npy_double, npy_longdouble,
+ *         npy_datetime, npy_timedelta#
+ * #isfloat = 0*10, 1*4, 0*2#
+ * #isnan = nop*10, npy_half_isnan, npy_isnan*3, nop*2#
+ * #lt = _LESS_THAN*10, _HALF_LESS_THAN, _LESS_THAN*5#
+ * #gt = _GREATER_THAN*10, _HALF_GREATER_THAN, _GREATER_THAN*5#
+ */
+static void
+@name@_fastclip(@type@ *in, npy_intp ni, @type@ *min, @type@ *max, @type@ *out)
+{
```

Comment:
Is this an exact copy of the `#else` branch? If you, that's not good - perhaps you can move the `HAVE_ATTRIBUTE_TARGET_AVX2` inside the function